### PR TITLE
boost 1.58.0

### PR DESCRIPTION
**Pull request by dunn**
closes #38743, which never got finished.

### CODE REVIEW

**Comment by dunn** [comment]
Fixed the checks locally; should I wait before queuing the build again?

**Comment by dunn** [comment]
```
        Error: 1 problems in 1 formulae
libswiften:
 * object files were linked against system openssl
These object files were linked against the deprecated system OpenSSL.
Adding `depends_on "openssl"` to the formula may help.
  /usr/local/Cellar/libswiften/2.0_1/lib/libSwiften.2.0.dylib
```
I guess it didn't help?

**Comment by DomT4** [comment]
Scons stuff usually requires a firm shove in the right direction. Add ` openssl=#{Formula["openssl"].opt_prefix} ` to the ` args ` list. Tested locally, works fine.

**Comment by dunn** [comment]
Should I wait for comments from the maintainers before running another build?

File: Library/Formula/boost.rb [hunk 1]
```diff
@@ -137,7 +137,6 @@ def caveats
       only supported on x86_64.
       EOS
     end
-
```
Comment (DomT4) [inline Library/Formula/boost.rb]: This line is usually left here in the ` caveats `. Not sure if that's deliberate, but it's usually there :smiley_cat:

File: Library/Formula/libswiften.rb [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 class Libswiften < Formula
   homepage 'http://swift.im/swiften'
-
+  revision 1
   stable do
```
Comment (DomT4) [inline Library/Formula/libswiften.rb]: Make sure there's an empty line between ` revision ` and ` stable `.

**Comment by DomT4** [comment]
If you wanted to, you could fix any errors raised by ` brew audit --strict ` but aside that, looks good to me. Essentially, ensure everything revisioned actually needs revisioned. I can't think of much else the maintainers would need to ping a discussion on, but I'm fairly frequently wrong :wink:.

**Comment by dunn** [comment]
OK, I wasn't sure if the protocol for revisions was to leave the rest of the formula untouched.  Since libswiften needed more help, though, I'll go ahead and audit it.

Yeah, I just revised the ones you suggested in #38743.  Did those all fail that time?

**Comment by DomT4** [comment]
> OK, I wasn't sure if the protocol for revisions was to leave the rest of the formula untouched

If there is, I've been breaking it as often as possible accidentally :grinning:.

> I just revised the formulae you suggested in #38743.

Forgot about my comment there. Yup, those failed :+1:.

**Comment by dunn** [comment]
Cool, thanks for the help!

File: Library/Formula/boost.rb (hunk 1 above)
Comment (dunn) [inline Library/Formula/boost.rb]: Oops, you're right.

File: Library/Formula/rcssserver.rb [hunk 3]
```diff
@@ -75,8 +74,7 @@ def install
   end
 
   test do
-    system "#{bin}/rcssserver help | head -1 | grep 'rcssserver-#{version}'"
-    system "#{bin}/rcsslogplayer --version | tail -1 | grep 'rcsslogplayer Version'"
-    system "#{bin}/rcssmonitor --version | tail -1 | grep 'rcssmonitor Version'"
+    assert_match version.to_s,
+                 shell_output("#{bin}/rcssserver help", 1)
```
Comment (jacknagel) [inline Library/Formula/rcssserver.rb]: Please don't break these assertions across multiple lines unless it is super long or something.

**Comment by dunn** [comment]
Shit, accidentally queued two builds because I was in a rush and pushed prematurely. Many apologies.

**Comment by dunn** [comment]
```
==> Installing ledger dependency: boost-python
==> Downloading https://downloads.sourceforge.net/project/boost/boost/1.58.0/boost_1_58_0.tar.bz2
Already downloaded: /Library/Caches/Homebrew/boost-python-1.58.0.tar.bz2
Error: boost was built in C++11 mode so boost-python must be built with --c++11.
```
Should `cxx11` be required for boost-python?

**Comment by DomT4** [comment]
@tdsmith's PR on `ledger` passed fine a couple of days ago so that'd fairly strongly implies the status quo is fine. I'm not quite sure why Boost is being built with `c++11` on Mavericks and Yosemite in this PR - It should only build `c++11` by default on 10.8 and below.

**Comment by tdsmith** [comment]
cpp-netlib has an improper (I think) unconditional dependency on `"boost" => "c++11"`.

**Comment by tdsmith** [comment]
This looks like it should be a safe change: http://upstream-tracker.org/compat_reports/boost/1.57.0_to_1.58.0/abi_compat_report.html -- a little surprised we need to revision anything.

**Comment by DomT4** [comment]
> cpp-netlib has an improper (I think) unconditional dependency on "boost" => "c++11".

@dunn Might be worth changing that in this PR and seeing if that kicks anything.

> a little surprised we need to revision anything.

Agreed, It looks good. Test bot had a field day breaking in the other PR though :cry:.

**Comment by dunn** [comment]
cpp-netlib built fine against non-cxx11 boost locally, so that's something.

**Comment by dunn** [comment]
But not on Mountain Lion.

**Comment by DomT4** [comment]
Oh, you probably need to do something like this to `cpp-netlib`:

```ruby
  if MacOS.version < :mavericks
    depends_on "boost" => "c++11"
  else
    depends_on "boost"
  end
```

**Comment by dunn** [comment]
Oh, yeah; I was thinking the boost formula itself might need fiddling but that looks right.

**Comment by DomT4** [comment]
Both `points2grid`, `libswiften` and `freeling` failed on ML with ` Undefined symbols for architecture x86_64: ` errors. `ledger`'s test failed.

Only ML for all of these errors. Mavericks and Yosemite are :100:.

**Comment by dunn** [comment]
But they passed in earlier builds, I thought?

**Comment by DomT4** [comment]
They did, yeah. Might be ` c++11 ` issues again given that ML was the only failure, but :confounded:.

**Comment by tdsmith** [comment]
The problem is related to how the test bot builds packages; packages built against libstdc++ need boost build against libstdc++ and packages built against libc++ need boost built against libc++. On Mavericks and newer, there's no problem since everything is build against libc++. On Mountain Lion, though, there's a mix. If a package needs a libc++ boost, `depends_on "boost" => "c++11"` means it will always get it, but there's no "without-c++11" option, so a package that needs a libstdc++ boost will sometimes fail if it's being built after a package that needed a libc++ boost.

I propose to pull this despite the Mountain Lion failures; the side effect is that this will remove the Mountain Lion bottles, but they can be rebottled separately. I assume the packages still work.

I see you added some new `boost => "c++11"` dependencies to troubleshoot; those should be removed.

**Comment by DomT4** [comment]
Tim's solution sounds good. Would be good to get someone using 10.8 to use the newly generated Boost bottles to check the other affected formulae in this PR do indeed work, but who is available and willing to test on a 10.8 machine isn't entirely obvious to me.

**Comment by dunn** [comment]
OK, reverted that.

**Comment by tdsmith** [comment]
ty! :fish_cake:

**Pull request closed without merging**